.github/workflows/reusable-test.yaml

```diff
@@ -113,3 +113,21 @@ jobs:
         uses: actions/upload-artifact@v3
         with:
           path: ./logs
+
+  codespell:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+
+      - name: Install cspell
+        run: npm install -g cspell
```

Comment:
You can replace this part with:

```yaml
      - name: Setup environment
        uses: ./.github/workflows/actions/setup-environment

      - name: Install dependencies
        uses: ./.github/workflows/actions/install-dependencies
```

Then add the `cspell` dependency as `devDependency` to the root `package.json` (either add it manually and run `pnpm install` or run `pnpm add -W -D cspell`). Don't forget to run `nvm use` before installing anything to make sure your node version matches the `.nvmrc`!